test/testdata/compiler/proc_arity.rb

```diff
@@ -24,6 +24,15 @@ def test_case(name, p)
 test_case("1 required, 1 defaulted", lambda {|a, b=nil|})
 test_case("1 required, 2 defaulted", lambda {|a, b=nil, c=nil|})
 test_case("2 required, 2 defaulted", lambda {|a, b, c=nil, d=nil|})
+test_case("2 required, 2 defaulted, break", lambda {|a, b, c=nil, d=nil| break})
+
+# keywords
+test_case("1 required", lambda {|a:|})
+test_case("2 required", lambda {|a:, b:|})
+test_case("1 optional", lambda {|a: 10|})
+test_case("2 optional", lambda {|a: 10, b: 20|})
+test_case("1 required, 1 optional", lambda {|a:, b: 20|})
+test_case("2 required, 1 optional", lambda {|a:, b:, c: 30|})
```

Comment:
Have testcases with positional and/or optional args + required and/or optional kwargs as well?

Reply:
Good suggestion, this exposed some problems!